File: 643. Maximum Average Subarray I/optimised.ts
function findMaxAverage(nums: number[], k: number): number {
	let window_sum: number = nums.slice(0, k).reduce((a, b) => a + b, 0);
	let res = window_sum;
	for (let i = k; i < nums.length; i++) {
		window_sum += nums[i] - nums[i - k];
		res = Math.max(window_sum, res);
	}

	return res / k;
};

console.log(findMaxAverage([1,12,-5,-6,50,3], 4));

File: 643. Maximum Average Subarray I/optimised.py
#!/usr/bin/env python

from typing import List

def findMaxAverage(nums: List[int], k: int) -> float:
    n = len(nums)
    window_sum = sum(nums[:k])
    res = window_sum
    for i in range(k, n):
        window_sum += nums[i] - nums[i - k]
        res = max(res, window_sum)

    return res / k

print(findMaxAverage([1,12,-5,-6,50,3], 4))
